### PR TITLE
Tagless AccountService

### PR DESCRIPTION
**Pull request by LordRaydenMK**
Algebra and interpeter for `AccountService`

### CODE REVIEW

File: src/main/kotlin/io/arrowkt/tagless/service/interpreter/AccountService.kt [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                Account.savingsAccount(no, name, it, openingDate, None, Balance())
+            )
+        }
+            .getOrElse { raiseError(IllegalArgumentException("Rate missing for savings account")) }
+    }
+
+    private fun createOrUpdate(
+        repo: AccountRepository<F>,
+        errorOrAccount: ErrorOr<Account>
+    ): Kind<F, Account> = when (errorOrAccount) {
```
Comment (pakoito) [inline src/main/kotlin/io/arrowkt/tagless/service/interpreter/AccountService.kt]: Use `fold` here, please.

File: src/main/kotlin/io/arrowkt/tagless/service/interpreter/AccountService.kt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        object C : DC()
+    }
+
+    override fun debit(no: String, amount: Amount) = update(no, amount, DC.D)
+
+    override fun credit(no: String, amount: Amount) = update(no, amount, DC.C)
+
+    private fun update(no: String, amount: Amount, debitCredit: DC): Kleisli<F, AccountRepository<F>, Account> =
+        Kleisli { repo ->
+            binding {
```
Comment (pakoito) [inline src/main/kotlin/io/arrowkt/tagless/service/interpreter/AccountService.kt:110]: Use `fx` if you can.

**Review by pakoito — APPROVED** [review]
Approved, couple of nits.

I'd suggest migrating to 0.9.1-SNAPSHOT and start using `fx` instead of `binding`

**Comment by LordRaydenMK** [comment]
I will do the migration once it hits stable.

File: src/main/kotlin/io/arrowkt/tagless/service/interpreter/AccountService.kt (hunk 1 above)
Comment (LordRaydenMK) [reply src/main/kotlin/io/arrowkt/tagless/service/interpreter/AccountService.kt]: Done

**Pull request merged**